src/server/scripts/Spells/spell_warlock.cpp

```diff
@@ -1147,31 +1147,27 @@ class spell_warl_seduction : public SpellScriptLoader
 };
 
 // -27285 - Seed of Corruption
-class spell_warl_seed_of_corruption : public SpellScriptLoader
+class spell_warl_seed_of_corruption : public SpellScript
 {
-    public:
-        spell_warl_seed_of_corruption() : SpellScriptLoader("spell_warl_seed_of_corruption") { }
+    PrepareSpellScript(spell_warl_seed_of_corruption);
 
-        class spell_warl_seed_of_corruption_SpellScript : public SpellScript
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        targets.remove_if([&](WorldObject const* target)
         {
-            PrepareSpellScript(spell_warl_seed_of_corruption_SpellScript);
-
-            void FilterTargets(std::list<WorldObject*>& targets)
-            {
-                if (GetExplTargetUnit())
-                    targets.remove(GetExplTargetUnit());
-            }
+            if (Unit const* unitTarget = target->ToUnit())
+                if (WorldLocation* dest = const_cast<WorldLocation*>(GetExplTargetDest()))
```

Comment:
do you need the const_cast for something in particular here ?

Reply:
copypaste from: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Northrend/IcecrownCitadel/boss_the_lich_king.cpp#L2294

Reply:
in the linked case the WorldLocation gets relocated, we're just getting the coords here => const_cast not required

Reply:
yep sorry, I couldn't make the change yesterday, thanks to both